code/controllers/subsystem/atoms.dm

```diff
@@ -85,30 +85,40 @@ SUBSYSTEM_DEF(atoms)
 	if (atoms_to_return)
 		LAZYINITLIST(created_atoms)
 
+	#ifdef TESTING
 	var/count
+	#endif
+
 	var/list/mapload_arg = list(TRUE)
 
 	if(atoms)
+		#ifdef TESTING
 		count = atoms.len
-		for(var/I in 1 to count)
+		#endif
+
+		for(var/I in 1 to atoms.len)
 			var/atom/A = atoms[I]
 			if(!(A.flags_1 & INITIALIZED_1))
 				CHECK_TICK
 				PROFILE_INIT_ATOM_BEGIN()
 				InitAtom(A, TRUE, mapload_arg)
 				PROFILE_INIT_ATOM_END(A)
 	else
+		#ifdef TESTING
 		count = 0
-		for(var/atom/A in world)
+		#endif
+
+		for(var/atom/A as anything in world)
```

Comment:
Did you check this actually iterates over everything. (Including creating new objects in init)  In world loops have some special handling.

Reply:
it looks ok to me when i orbited and i hope itll fail a unit test